src/west/app/project.py

```diff
@@ -924,6 +924,29 @@ def fetch_strategy(self, args):
         else:
             return 'smart'
 
+    def update_submodules(self, project, submodules, call_location):
+        # Updates given project submodules by using
+        # 'git submodule update --init --checkout --recursive' command
+        # from the location passed through the call_location parameter value.
+        if submodules is not None:
```

Comment:
You can save one indentation level with an early return:
```python
if submodules is None:
   return
```

Reply:
With a `all_submodules` boolean and a possibly empty submodules list you would not even need this `if` at all: iterating over a possibly empty list is perfectly fine.

Reply:
I applied the version from the first comment.